core/src/main/java/io/grpc/internal/InternalSubchannel.java

```diff
@@ -221,21 +205,24 @@ private Runnable startNewTransport() {
     }
     pendingTransport = transport;
     transports.add(transport);
-    return transport.start(new TransportListener(transport, address));
+    Runnable runnable = transport.start(new TransportListener(transport, address));
+    if (runnable != null) {
+      channelExecutor.executeLater(runnable);
```

Comment:
Per our offline discussion, I have applied an easier-to-audit convention to make sure `ChannelExecutor` is drained. PTAL.